pkg/initializer/initializer.go

```diff
@@ -64,23 +64,125 @@ const (
 	listKind        = "StorageVersionMigrationList"
 )
 
-func migrationCRD() *apiextensionsv1beta1.CustomResourceDefinition {
-	return &apiextensionsv1beta1.CustomResourceDefinition{
+func migrationCRD() *v1.CustomResourceDefinition {
+	return &v1.CustomResourceDefinition{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("%s.%s", pluralCRDName, migrationv1alpha1.GroupName),
+			Name: "storageversionmigrations.migration.k8s.io",
+			Annotations: map[string]string{
+				"api-approved.kubernetes.io": "https://github.com/kubernetes/community/pull/2524",
+			},
 		},
-		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
-			Group:   migrationv1alpha1.GroupName,
-			Version: migrationv1alpha1.SchemeGroupVersion.Version,
-			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+		Spec: v1.CustomResourceDefinitionSpec{
+			Group: "migration.k8s.io",
+			Names: v1.CustomResourceDefinitionNames{
 				Plural:   pluralCRDName,
 				Singular: singularCRDName,
 				Kind:     kind,
 				ListKind: listKind,
 			},
-			Scope: apiextensionsv1beta1.ClusterScoped,
-			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
-				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			Scope: v1.ClusterScoped,
+			Versions: []v1.CustomResourceDefinitionVersion{
+				{
+					Name:    "v1alpha1",
+					Served:  true,
+					Storage: true,
+					Subresources: &v1.CustomResourceSubresources{
+						Status: &v1.CustomResourceSubresourceStatus{},
+					},
+					Schema: &v1.CustomResourceValidation{
```

Comment:
is the initializer or the manifest used to create the CRD? does this mean it wasn't being created with a schema before?

Reply:
The initializer should be deprecated. It's only used in the e2e test. We need to remove it and update the relevant e2e test in the future.